src/Raven.Server/Utils/TypeConverterBase.cs

```diff
@@ -24,26 +22,136 @@
 
-        private static readonly ConcurrentDictionary<Type, IPropertyAccessor> PropertyAccessorCache = new ConcurrentDictionary<Type, IPropertyAccessor>();
+        protected static readonly StringSegment ValuePropertyName = new StringSegment("$value");
 
-        private static readonly ConcurrentDictionary<Type, IPropertyAccessor> PropertyAccessorForMapReduceOutputCache = new ConcurrentDictionary<Type, IPropertyAccessor>();
+        protected static readonly ConcurrentDictionary<Type, IPropertyAccessor> PropertyAccessorCache = new ConcurrentDictionary<Type, IPropertyAccessor>();
 
-        private static readonly TypeCache<bool> _isSupportedTypeCache = new (512);
+        protected static readonly ConcurrentDictionary<Type, IPropertyAccessor> PropertyAccessorForMapReduceOutputCache = new ConcurrentDictionary<Type, IPropertyAccessor>();
+
+        protected static readonly TypeCache<bool> _isSupportedTypeCache = new (512);
+
+        public TypeConverterBase(JavaScriptEngineType jsEngineType)
+        {
+            _jsEngineType = jsEngineType;
+        }
+        
+        public static JavaScriptEngineType GetJsEngineType(IJavaScriptOptions jsOptions)
+        { 
+            return GetJsEngineType(jsOptions?.EngineType); 
+        }
+        
+        public static JavaScriptEngineType GetJsEngineType(JavaScriptEngineType? jsEngineType)
+        { 
+            return jsEngineType ?? JavaScriptEngineType.Jint; 
+        }
+        
+        public static IJsTypeConverter GetTypeConverter(JavaScriptEngineType? jsEngineType)
+        {
+            return GetJsEngineType(jsEngineType) switch
+            {
+                JavaScriptEngineType.Jint => TypeConverterJint,
+                JavaScriptEngineType.V8 => TypeConverterV8,
+                _ => throw new NotSupportedException($"Not supported JS engine type '{jsEngineType}'.")
+            };
+        }
+        
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static object ToBlittableSupportedType(JavaScriptEngineType? jsEngineType, object value, bool flattenArrays = false, bool forIndexing = false, object engine = null,
+            JsonOperationContext context = null, bool isRoot = true)
+        {
+            var typeConverter = GetTypeConverter(jsEngineType);
+            return typeConverter.ToBlittableSupportedType(value, flattenArrays, forIndexing, engine, context, isRoot);
+        }
 
-        private static bool IsSupportedTypeInternal(object value)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static object ToBlittableSupportedType(JavaScriptEngineType? jsEngineType, object value, out BlittableSupportedReturnType returnType, bool flattenArrays = false, bool forIndexing = false,
+            object engine = null, JsonOperationContext context = null)
+        {
+            var typeConverter = GetTypeConverter(jsEngineType);
```

Comment:
This has been a *really* invasive change, but I don't see why we need it.

Reply:
I have merged Jint and V8 versions and simplified TypeConverter class, turned it back to be static, and eliminated jsEngineType arguments in ToBlittableSupportedType and IsSupportedType